### PR TITLE
Add shared Jenkins library

### PR DESCRIPTION
**Pull request by markysoft**
https://eaflood.atlassian.net/browse/PSD-219

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,137 +1,32 @@
...
     }
     stage('Run tests') {
-      runTests(imageName, BUILD_NUMBER)
+      defraUtils.runTests(imageName, BUILD_NUMBER)
     }
-    // note: there should be a `build production image` step here,
-    // but the docker file is currently not set up to create a production only image
     stage('Push container image') {
-      pushContainerImage(registry, regCredsId, imageName, containerTag)
+      defraUtils.pushContainerImage(regCredsId, registry, imageName, containerTag)
```
Comment (Matthew-Collins-Defra) [inline Jenkinsfile]: This should really be called "buildAndPushImage" instead, as it's not obvious that the build step is in here.

File: Jenkinsfile [hunk 2]
```diff
@@ -141,26 +36,24 @@ node {
...
     }
     if (mergedPrNo != '') {
       stage('Remove merged PR') {
-        sh "echo removing deployment for PR $mergedPrNo"
-        undeployPR(kubeCredsId, imageName, mergedPrNo)
+        defraUtils.undeployChart(kubeCredsId, imageName, mergedPrNo)
       }
     }
-    updateGithubCommitStatus('Build successful', 'SUCCESS', repoUrl, commitSha)
+    defraUtils.updateGithubCommitStatus('Build successful', 'SUCCESS')
```
Comment (Matthew-Collins-Defra) [inline Jenkinsfile]: I've been considering if we should have 3 methods for this, instead of passing string statuses in, I'm not keen on magic values.

Comment (markysoft) [reply Jenkinsfile]: Good point, there are some other statuses too, like UNKNOWN that we wouldn't use. 
The fewer parameters the better.

Might be worthwhile doing it now before all the other branches are merged. i.e. leave the `updateGithubCommitStatusupdate` as it is but add `setGithubStatusSuccess`, `setGithubStatusPending` and `setGithubStatusFailure`

Comment (Matthew-Collins-Defra) [reply Jenkinsfile]: Good idea. I've created a new branch to do that, and I'm testing against the user-service.

Comment (markysoft) [reply Jenkinsfile]: updated code to use new github methods

File: Jenkinsfile [hunk 3]
```diff
@@ -1,137 +1,32 @@
+@Library('defra-library@master')
```
Comment (Matthew-Collins-Defra) [inline Jenkinsfile]: Should this be using the tagged version of the library?

**Pull request merged**